### PR TITLE
Remove guzzle parser

### PR DESCRIPTION
**Pull request by hiend**
Simple response parser

### CODE REVIEW

File: src/ResponseParser.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+        list($http, $code, $reason) = explode(' ', array_shift($lines), 3);
+        list($protocol, $version) = explode('/', $http, 2);
+
+        $headers = [];
+        foreach ($lines as $line) {
+            list($header, $value) = array_map('trim', explode(':', $line, 2));
+
+            if (!isset($headers[$header])) {
+                $headers[$header] = $value;
```
Comment (cboden) [inline src/ResponseParser.php]: IMO I think it would be better if `$value` was always an array. This would prevent the consumer from having to type check.

**Comment by cboden** [comment]
A good start! Off the top of my head this is also a valid multi value header:
> A-Header: value1,value2,value3

**Comment by hiend** [comment]
hmm, values may contain comma for a different reasons:
> Last-Modified: Wed**`,`** 28 Apr 2015 11:20:59 GMT

**Comment by WyriHaximus** [comment]
Aren't header values seperated by `;` and not `,`?

**Comment by hiend** [comment]
github request :)
```
HTTP/1.1 200 OK
Server: GitHub.com
Date: Fri, 01 May 2015 18:47:47 GMT
Content-Type: text/html; charset=utf-8
Transfer-Encoding: chunked
Status: 200 OK
Content-Security-Policy: blablabla
Cache-Control: no-cache
X-GitHub-User: hiend
X-GitHub-Session-Id: blablabla
Vary: X-PJAX
X-UA-Compatible: IE=Edge,chrome=1
Set-Cookie: user_session=blablabla; path=/; expires=Fri, 15-May-2015 18:47:47 GMT; secure; HttpOnly
Set-Cookie: _gh_sess=blablabla; path=/; secure; HttpOnly
X-Request-Id: blabla
X-Runtime: 0.238943
X-GitHub-Request-Id: bla:bla:bla:bla
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
X-Frame-Options: deny
Vary: Accept-Encoding
X-Served-By: blabla
Content-Encoding: gzip
```

**Comment by WyriHaximus** [comment]
:+1: 

Also we would very highly appreciate unit tests.

File: src/ResponseParser.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+        list($http, $code, $reason) = explode(' ', $first_line.' ', 3);
+
+        if (!strpos($http, '/') or (int)$code < 100 or (int)$code >= 1000) {
+            return false;
+        }
+
+        list($protocol, $version) = explode('/', $http, 2);
+
+        if ($protocol !== 'HTTP' or $version !== '1.0') {
```
Comment (WyriHaximus) [inline src/ResponseParser.php]: Not sure about the pinning down on 1.0. Using it to talk to 1.1 servers a lot and that works fine. Not sure what their responses are but blocking 1.1 responses while they used to work in previous 0.4.x versions could be a BC break. Will investigate for you :).

**Comment by WyriHaximus** [comment]
I'm closing this issue in favor of #34. You've done a great job but after @cboden https://github.com/reactphp/http/pull/29 I've realized that using an external package that is specialized in message parsing is the better way to go. Your work is much appreciated and this decision isn't because of the work you've put into it. Once again thank you :+1:

**Pull request closed without merging**